genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncher.java

```diff
@@ -0,0 +1,226 @@
+        final Cache<String, String> healthIndicatorCache,
+        final GenieHostInfo genieHostInfo,
+        final TitusAgentLauncherProperties titusAgentLauncherProperties,
+        final MeterRegistry registry
+    ) {
+        this.restTemplate = restTemplate;
+        this.healthIndicatorCache = healthIndicatorCache;
+        this.genieHostInfo = genieHostInfo;
+        this.titusAgentLauncherProperties = titusAgentLauncherProperties;
+        this.registry = registry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<JsonNode> launchAgent(
+        final ResolvedJob resolvedJob,
+        @Nullable final JsonNode requestedLauncherExt
+    ) throws AgentLaunchException {
+
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        final TitusBatchJobRequest titusJobRequest = this.createJobRequest(resolvedJob);
+        final Set<Tag> tags = Sets.newHashSet();
+
+        String titusJobId = "unknown";
+
+        final long start = System.nanoTime();
+        try {
+            final TitusBatchJobResponse titusResponse = this.restTemplate.postForObject(
+                this.titusAgentLauncherProperties.getEndpoint(),
+                titusJobRequest,
+                TitusBatchJobResponse.class
+            );
+
+            if (titusResponse == null) {
+                throw new AgentLaunchException("Failed to request creation of Titus job for job " + jobId);
+
+            } else if (StringUtils.isBlank(titusResponse.getId())) {
+                throw new AgentLaunchException(
+                    "Failed to create titus job for job " + jobId
+                        + " - " + titusResponse.getStatusCode() + ": "
+                        + titusResponse.getMessage()
+                );
+
+            }
+
+            titusJobId = titusResponse.getId();
+
+            log.info("Created Titus job {} to execute job {}", titusJobId, jobId);
+
+            MetricsUtils.addSuccessTags(tags);
+
+            final JsonNode launcherExt = JsonNodeFactory.instance.objectNode()
+                .put("launcher_class", this.getClass().getCanonicalName())
+                .put("titus_job_id", titusJobId)
+                .putPOJO("titus_job_request", titusJobRequest)
+                .putPOJO("titus_job_response", titusResponse);
+
+            return Optional.of(launcherExt);
+
+        } catch (Exception e) {
+            log.error("Failed to launch job on Titus", e);
+            MetricsUtils.addFailureTagsWithException(tags, e);
+            throw new AgentLaunchException("Failed to create titus job for job " + jobId, e);
+        } finally {
+            final long end = System.nanoTime();
+            this.registry.timer(LAUNCH_TIMER, tags).record(end - start, TimeUnit.NANOSECONDS);
+            this.healthIndicatorCache.put(jobId, titusJobId);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Health health() {
+        return Health.up()
```

Comment:
I think longer term we'd want to do something that maybe pinged the Titus health endpoint periodically for its health rather than always returning up

Reply:
Agreed. This launcher has no business being a health indicator, but it needs to be because `AgentLauncher` interface requires it. This kind of info belongs into an `InfoContributor`. 
I may detangle them in a later commit.